bot/cogs/Calculator.py

```diff
@@ -0,0 +1,153 @@
+import logging
+import re
+import discord
+import discord.ext.commands as commands
+
+log = logging.getLogger(__name__)
+
+
+class Calculator(commands.Cog):
+
+    def __init__(self, bot):
+        self.bot = bot
+        self._last_member = None
+        self.operators = [
+            {'symbol': '+', 'precedence': 0, 'assoc': 'L'},
+            {'symbol': '-', 'precedence': 0, 'assoc': 'L'},
+            {'symbol': '*', 'precedence': 1, 'assoc': 'L'},
+            {'symbol': '/', 'precedence': 1, 'assoc': 'L'},
+            {'symbol': '^', 'precedence': 2, 'assoc': 'R'},
+        ]
+
+    @commands.command()
+    async def calc(self, ctx, *args):
+        await ctx.send('{} arguments: {}'.format(len(args), ', '.join(args)))
+
+        result = self.parse_postfix(self.parse_expression(args[0]))
```

Comment:
calc should support any number of arbitrary spaces in the input for example $calc 2 +         2 , you should join args into one string before sending it to the parse function, this is also where you should do your input validation